morphia/src/main/java/org/mongodb/morphia/CollationBuilder.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mongodb.morphia;
+
+import com.mongodb.client.model.CollationAlternate;
+import com.mongodb.client.model.CollationCaseFirst;
+import com.mongodb.client.model.CollationMaxVariable;
+import com.mongodb.client.model.CollationStrength;
+import org.mongodb.morphia.annotations.Collation;
+
+class CollationBuilder extends AnnotationBuilder<Collation> implements Collation {
+    @Override
+    public Class<Collation> annotationType() {
+        return Collation.class;
+    }
+
+    @Override
+    public boolean backwards() {
+        return get("backwards");
```

Comment:
I didn't know the compiler could be used to set the result type for the get method. Pure voodoo!